beancount/parser/lexer.l

```diff
@@ -382,7 +389,7 @@ NULL		{
 
  /* Currencies. These are defined as uppercase only in order to disambiguate the
   * syntax. This is kept in sync with beancount.core.amount.CURRENCY_RE. */
-[A-Z][A-Z0-9\'\.\_\-]{0,22}[A-Z0-9]	{
```

Comment:
I actually need to add one to the maximum length here in order to preserve the 24 character limit that is noted in the documentation.  But is there a reason for that limit?